Bixal/move.mil
# Hot fix: PPM tool is wrong when zip3 rate area is 'ZIP'

Thanks for submitting a pull request! Below are a few things you can do to help us more quickly review your changes.

## Checklist

I have…

      against my changes

## Summary of Changes

This pull request…

- Fixes 508 compliance failures.
- Changes the query to the zip5 table. Instead of fetching the column rate_area which does not exist, now it is fetching the correct column name service_area.
- Cast to float the pack and unpack array values instead of the array. Casting the array to float is making the array null. Then the pack and unpack values are equal to 0. That was clearly a mistake, I meant to cast the values from the array. It's fixed now.

## Testing

To verify the changes proposed in this pull request…

1.  Pull branch changes and clear caches.
1.  Make sure tools are working.
1.  Use the PPM tool with the following data
```
{
    "selectedEntitlement": "o-3-w-3",
    "isDependencies": true,
    "locations": {
        "origin": "93230",
        "destination": "78236"
    },
    "weightOptions": {
        "houseHold": "6000",
        "proGear": "0",
        "dependent": "0"
    },
    "selectedMoveDate": "2018-08-22T17:00:45.527Z"
}
```

The results must be
```
{
    "incentive": {
        "min": 4600,
        "max": 5300
    },
    "advancePayment": {
        "min": 2760,
        "max": 3180,
        "percentage": 60
    
}
```

## Screenshots

<img width="504" alt="screen shot 2018-08-15 at 10 03 45 pm" src="https://user-images.githubusercontent.com/37001835/44182970-242ad600-a0d7-11e8-8e8e-1ef62fd9ad1d.png">
